src/containers/Auth/ResetPassword/UpdatePassword.js

```diff
@@ -0,0 +1,212 @@
+  static propTypes = {
+    toastManager: PropTypes.object.isRequired,
+    response: PropTypes.string,
+    error: PropTypes.bool,
+    updatePassword: PropTypes.func,
+    isLoading: PropTypes.bool,
+    token: PropTypes.string,
+    history: PropTypes.object,
+    passwordChanged: PropTypes.bool,
+  }
+
+  /**
+   * @constructor
+   * @description holds the state;
+   */
+  constructor() {
+    super();
+    this.state = {
+      error: {},
+      token: '',
+      data: {
+        password: '',
+        comfirmPassword: '',
+      },
+    };
+  }
+
+  componentDidMount = () => {
+    const { history: { location: { search } } } = this.props;
+    if (!search) {
+      this.props.history.push('/password-reset');
+    }
+  }
+
+  /**
+   * @param {*} e - event that is acted upon
+   * @returns {object} - The changed state
+   */
+  handleInputChange = (e) => {
+    const password = { type: e.target.id, content: e.target.value };
+    const errorState = this.state.error;
+    const isError = validate(password, errorState);
+    this.setState({
+      data: {
+        ...this.state.data,
+        [e.target.id]: e.target.value,
+      },
+      error: isError,
+    });
+  }
+
+
+  /**
+   * @description - decides if component should throw error or update
+   * @param {*} nextProps
+   * @param {*} nextState
+   * @returns {bool} - if the component should be uppdated or not
+   */
+  shouldComponentUpdate(nextProps) {
+    if (this.props.error !== nextProps.error && nextProps.error === true) {
+      this.props.toastManager.add(`${nextProps.response}`, {
+        appearance: 'error',
+        autoDismiss: true,
+      });
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   *
+   * @param {*} e The event that is to be submitted
+   * @returns {func} - calls the send email function
+   */
+  handleSubmit = (e) => {
+    const { history: { location: { search } } } = this.props;
+    const userToken = search.split('?')[1];
+    e.preventDefault();
+    const {
+      props: { toastManager }, state: { error, password, confirmPassword },
+    } = this;
+    if (error.password) {
+      toastManager.add('Invalid password supplied', {
+        appearance: 'error',
+        autoDismiss: true,
+      });
+      return false;
+    }
+    if (password !== confirmPassword) {
+      toastManager.add('Password must be equal to confirm password', {
+        appearance: 'error',
+        autoDismiss: true,
+      });
+      return false;
+    }
+    const { data } = this.state;
+    this.props.history.replace('/password-update');
+    this.setState({
+      ...data,
+      token: userToken,
+    }, () => {
+      this.props.updatePassword(this.state.data, this.state.token);
+    });
+  }
+
+  /**
+   * @function
+   * @returns {JSX} The update password form JSX
+   * or the update password successful JSX
+   */
+  render() {
+    return (
+        <FormWrapper imageUrl={image}
+        imgId='updateSideImage' displayImage={false}>
```

Comment:
can you change the prop name to `imageId` or something more readable

Reply:
Thank you for the feedback. imgId was added as a prop on the Formwrapper component. Changing it now would affect all the files that use the component. We would take note of this during refactoring.

Reply:
@idmega2000, you can change that, we only have two features merged